### PR TITLE
RFC: Web Servers Re-Architecture

### PR DESCRIPTION
**Pull request by ForestEckhardt**
[Readable](https://github.com/paketo-buildpacks/rfcs/blob/web-servers-rearch/text/web-servers/0006-web-servers-rearchitecture.md)

### CODE REVIEW

**Review by TisVictress — APPROVED** [review]

File: text/web-servers/0006-web-servers-rearchitecture.md
Comment (ryanmoran) [inline text/web-servers/0006-web-servers-rearchitecture.md:21]:
```suggestion
### {web-server}
```

Why do we feel the need to tack on `-dist` here?

**Comment by joshuatcasey** [comment]
2022-07-26 Working Group notes: waiting for feedback.

Should have minimal impact on the PHP language family that consumes `httpd`/`nginx` because PHP will likely only want the `-dist` buildpack.

File: text/web-servers/0006-web-servers-rearchitecture.md
Comment (ForestEckhardt) [reply text/web-servers/0006-web-servers-rearchitecture.md:21]: To differentiate it from the current `nginx` or `httpd` buildpacks which have the whole package

Comment (ryanmoran) [reply text/web-servers/0006-web-servers-rearchitecture.md:21]: Would it not be reasonable to keep those names for the distribution buildpacks and bump the major version in the process?

Comment (ForestEckhardt) [reply text/web-servers/0006-web-servers-rearchitecture.md:21]: @paketo-buildpacks/web-servers-maintainers Can I get a temperature check on this idea?

**Comment by dmikusa-pivotal** [comment]
2022-08-02 WG Notes:

Pending a bit more feedback from other web servers maintainers. There is a question about naming, using `-dist` or not. If no objections to the proposed naming then @ForestEckhardt will merge.

**Review by thitch97 — APPROVED** [review]

**Comment by davidmirror-ops** [comment]
WG 08-09-2022 Notes:
More approvals are required.
TH: there's a constant mention about the layer limit for Docker, would that still be a concern with this proposal?
EC: I think we need isolation. Pretty much anything we add is a new layer
TH: Agree
DM: it might be nice to know where we are in # of layers to see if we are close to the limit

**Comment by fg-j** [comment]
As of v0.2.151, the full builder has 117 layers. The docker limit is 125.
```
docker pull paketobuildpacks/builder:full
docker history paketobuildpacks/builder:full | wc -l
     117
```

File: text/web-servers/0006-web-servers-rearchitecture.md
Comment (ekcasey) [inline text/web-servers/0006-web-servers-rearchitecture.md:15]: Would it be possible to elaborate a big more on the motivation? Why is this an issue for PHP? What use cases are impossible to meet now that would be enabled by this re-architecture?

File: text/web-servers/0006-web-servers-rearchitecture.md
Comment (ekcasey) [inline text/web-servers/0006-web-servers-rearchitecture.md:265]: What would it look like to "web server dependencies are ingestible by third party buildpacks" without this refactor?

**Comment by joshuatcasey** [comment]
2022-08-16 Working Group Notes: @ekcasey has left comments (see above), those comments should be addressed by next week as per @ForestEckhardt

File: text/web-servers/0006-web-servers-rearchitecture.md
Comment (ForestEckhardt) [reply text/web-servers/0006-web-servers-rearchitecture.md]: I think I have a more solid paragraph explaining this now but let me know if it is still unclear.

**Comment by ForestEckhardt** [comment]
WG 2022-08-23 Notes:
Needs response from @ekcasey after update.

File: text/web-servers/0006-web-servers-rearchitecture.md
Comment (ekcasey) [reply text/web-servers/0006-web-servers-rearchitecture.md]: Thanks for the added context!

**Review by ekcasey — APPROVED** [review]

**Comment by davidmirror-ops** [comment]
WG 2022-08-30 Notes: OK to merge

**Pull request merged**
